src/main/plugins/uwp-plugin/uwp-plugin.ts

```diff
@@ -4,38 +4,40 @@ import { UserConfigOptions } from "../../../common/config/user-config-options";
 import { TranslationSet } from "../../../common/translation/translation-set";
 import { PluginType } from "../../plugin-type";
 import { UwpApplication } from "./uwp-application";
+import { UwpAppsRetriever } from "./uwp-apps-retriever";
 import { UwpSearchOptions } from "../../../common/config/uwp-search-options";
-import { UwpAppRepository } from "./uwp-app-repository";
 
 export class UwpPlugin implements SearchPlugin {
     public pluginType = PluginType.Uwp;
     private config: UwpSearchOptions;
-    private readonly uwpAppRepository: UwpAppRepository;
+    private uwpApps: UwpApplication[];
     private readonly filePathExecutor: (executionArgument: string, privileged: boolean) => Promise<void>;
 
     constructor(
         config: UwpSearchOptions,
-        uwpAppRepository: UwpAppRepository,
         filePathExecutor: (executionArgument: string, privileged: boolean) => Promise<void>,
-        ) {
+    ) {
         this.config = config;
-        this.uwpAppRepository = uwpAppRepository;
         this.filePathExecutor = filePathExecutor;
+        this.uwpApps = [];
     }
 
     public getAll(): Promise<SearchResultItem[]> {
+        return new Promise((resolve) => resolve(this.uwpApps.map((app) => this.createSearchResult(app))));
+    }
+    public refreshIndex(): Promise<void> {
         return new Promise((resolve, reject) => {
-            this.uwpAppRepository.getAll()
-                .then((uwpApps) => {
-                    resolve(uwpApps.map((uwpApp) => this.createSearchResult(uwpApp)));
+            UwpAppsRetriever
+                .getAll(this.uwpApps)
+                .then((apps) => {
+                    this.uwpApps = apps;
+                    resolve();
                 })
-                .catch((err) => reject(err));
+                .catch((error) => reject(error));
```

Comment:
I'd like to have the UwpPlugin not depend on UwpAppsRetriever. How about having the UwpPlugin depend on a function that returns all UWP apps (similar to the `filePathExecutor`). The advantage with this solution would be that the UwpPlugin class could be unit tested much easier in a later stage because it does not depend on the hardcoded UwpAppRetriever class.

Here is an example:

uwp-plugin.ts
``` typescript
class UwpPlugin {
    private uwpAppsRetriever: () => Promise<UwpApplication[]>;

    constructor(uwpAppsRetriever: () => Promise<UwpApplication[]>) {
        this.uwpAppsRetriever = uwpAppsRetriever;
    }

    public getAll(): Promise<SearchResultItem[]> {
        return new Promise((resolve, reject ) => {
            this.uwpAppsRetriever()
                .then((uwpApps) => resolve(this.createSearchResults(uwpApps)))
                .catch((error) => reject(error));
        })
    }

    private createSearchResults(uwpApps: UwpApplication[]): SearchResultItem[] {
        return uwpApps.map((uwpApp): SearchResultItem => {
            return {
                // create search result item here
            };
        })
    }
}
```

uwp-apps-retriever.ts
``` typescript
function convertNewAppToUwpApplication() {
    // stuff from "private static convertNewAppToUwpApplication" goes here
}

function executeCommandWithUtf8Output() {
    // stuff from "private static executeCommandWithUtf8Output" goes here
}

const uwpAppsPowershellCommand = ``; // powershell command goes here

export function getAllUwpApps(): Promise<UwpApplication[]> {
    // get all uwp apps here
}
```

production-search-engine.ts
``` typescript
import { getAllUwpApps } from "./uwp-apps-retriever";

searchPlugins.push(new UwpPlugin(getAllUwpApps));
```

Reply:
You probably meant to call the uwpAppsRetriever in refreshIndex(), didn't you?

Reply:
Sorry, yes :)

Reply:
Alright, that makes sense. I pushed the change.